pkg/provider/gcp/secretmanager/client.go

```diff
@@ -129,46 +162,38 @@ func parseError(err error) error {
 
 // PushSecret pushes a kubernetes secret key into gcp provider Secret.
 func (c *Client) PushSecret(ctx context.Context, payload []byte, remoteRef esv1beta1.PushRemoteRef) error {
-	createSecretReq := &secretmanagerpb.CreateSecretRequest{
-		Parent:   fmt.Sprintf("projects/%s", c.store.ProjectID),
-		SecretId: remoteRef.GetRemoteKey(),
-		Secret: &secretmanagerpb.Secret{
-			Labels: map[string]string{
-				"managed-by": "external-secrets",
-			},
-			Replication: &secretmanagerpb.Replication{
-				Replication: &secretmanagerpb.Replication_Automatic_{
-					Automatic: &secretmanagerpb.Replication_Automatic{},
-				},
-			},
-		},
-	}
-
-	var gcpSecret *secretmanagerpb.Secret
-	var err error
-
-	gcpSecret, err = c.smClient.GetSecret(ctx, &secretmanagerpb.GetSecretRequest{
+	gcpSecret, err := c.smClient.GetSecret(ctx, &secretmanagerpb.GetSecretRequest{
 		Name: fmt.Sprintf("projects/%s/secrets/%s", c.store.ProjectID, remoteRef.GetRemoteKey()),
 	})
 	metrics.ObserveAPICall(constants.ProviderGCPSM, constants.CallGCPSMGetSecret, err)
 
-	var gErr *apierror.APIError
+	if err != nil {
+		if remoteRef.GetProperty() != "" || status.Code(err) != codes.NotFound {
```

Comment:
Thank you so much for your detailed feedback, @moolen! While waiting for a response from @gusfcarvalho, I've updated the PR in the following ways 🙂 

- Create a SecretManager secret if it does not exist
- Delete a whole SecretManager secret even when the property is specified
- If a property is specified and the managed-by label is missing, attach the label and take over the ownership

Reply:
> vault: [takes over ownership](https://github.com/external-secrets/external-secrets/blob/bdf437c2e1a1a441955b1036b84eed5f63bda5d9/pkg/provider/vault/vault.go#L523)

While reading the source code, I noticed [valut actually does not seem to take over the ownership](https://github.com/external-secrets/external-secrets/blob/69fe93ea49ca078b791ce8b00094fd9defa60da6/pkg/provider/vault/vault.go#L477-L486)and returns an error if the appropriate managed-by label is not attached 🤔

Reply:
Sound good to me!
Sorry, i got carried away with other stuff the last days (weeks :sweat_smile:) - @gusfcarvalho do you have input on this one? Otherwise i'd do a final round of testing. So far the changes proposed LGTM.